Algorithm/QCAlgorithm.History.cs

```diff
@@ -42,7 +42,7 @@ public IHistoryProvider HistoryProvider
         public bool IsWarmingUp
         {
             get;
-            private set;
+            internal set;
```

Comment:
Is this necessary? If you need to transition the algorithm from `IsWarmingUp==true` to `IsWarmingUp==false` you can use `SetFinishedWarmingUp()`. Ideally we don't need to use internal modifiers to enable testing.